client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx

```diff
@@ -0,0 +1,211 @@
+import { Typography, Button, TextField, Box, CircularProgress, Grid } from '@material-ui/core';
+import { Formik, FormikHelpers, useFormik, validateYupSchema } from 'formik';
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import { useAuth } from '../../../../context/useAuthContext';
+
+//This form needs to UPDATE to /profile/edit/update
+
+export default function ProfileEditForm(): JSX.Element {
+  const { loggedInUser } = useAuth();
+  const [state, setState] = useState({
+    firstName: '',
+    lastName: '',
+    email: '',
+    gender: '',
+    birthday: '',
+    phone: '',
+    address: '',
+    description: '',
+  });
+
+  const [hasProfile, setHasProfile] = useState(false);
+
+  useEffect(() => {
+    async function fetchProfile() {
+      await axios
+        .get('/profile')
+        .then((response: any) => {
+          if (typeof response.data.profile === 'object') {
+            const profile = response.data.profile;
+            if (profile === null) {
+              setHasProfile(false);
+            } else {
+              setState(profile);
+              setHasProfile(true);
+            }
+          }
+        })
+        .catch((error) => {
```

Comment:
I'd suggest you make proper use of `async/await` to simplify your code and make it more readable.